Modules/IO/ImageBase/include/itkImageIOBase.h

```diff
@@ -710,8 +741,10 @@ class ITKIOImageBase_EXPORT ImageIOBase : public LightProcessObject
   /** Does the ImageIOBase object have enough info to be of use? */
   bool m_Initialized{};
 
+#ifndef ITK_FUTURE_LEGACY_REMOVE
   /** Filename to read */
-  std::string m_FileName{};
+  const std::string m_FileName{};
+#endif
```

Comment:
It would be preferable to make `m_FileName` private, instead of protected. Then the workaround (`const_cast<std::string &>`) would not be necessary at all. ImageIOBase has a GetFileName and a SetFileName anyway, so direct access to `m_FileName` should not be necessary.